client/consensus/common/src/parachain_consensus.rs

```diff
@@ -210,7 +190,7 @@ async fn follow_new_best<P, R, Block, B>(
 						h,
 						&*parachain,
 						&mut unset_best_header,
-						recovery_chan_tx.clone(),
+						recovery_chan_tx.as_mut(),
```

Comment:
As we clonded the sender every time before, the `try_send` would have always succeeded as each sender instance gets at least one slot in the channel.